forte/processors/base/data_augment_processor.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processors that augment the data.
+"""
+from abc import abstractmethod
+from forte.common.resources import Resources
+from forte.common.configuration import Config
+from forte.processors.data_augment.algorithms.dictionary_replacement_augmenter import DictionaryReplacementAugmenter
```

Comment:
Now you are importing a specific Augmenter into the parent Augmenter, this will probably cause cyclic import. We shouldn't have concrete implementations here.

Reply:
The parent is a processor, while the child is an augmenter. They are two types of classes.

Reply:
Yeah, but we are still adding all the specific augmented instance to a base class. That means we have to change the code of the base class everytime we have a new algorithm, which does not sound right.